### PR TITLE
Implement EntityInteractionType methods

### PR DESCRIPTION
**Pull request by Aaron1011**

### CODE REVIEW

File: src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            return false;
+        }
+        SpongeEntityInteractionType other = (SpongeEntityInteractionType) obj;
+        if (this.name != other.name) {
+            return false;
+        }
+        return true;
+    }
+
+    public DataContainer toContainer() {
```
Comment (simon816) [inline src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java]: What are these serialization methods doing here?

Comment (Aaron1011) [inline src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java]: They're present in `SpongeEntityMeta`, which has the same purpose as this class (except it also requires a `type`, something unnecessary for this).

Those methods should be removed from `SpongeEntityMeta` as well if they don't belong here.

Comment (simon816) [inline src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java]: I don't see why these methods are here or in `SpongeEntityMeta`, if there's no `@Override` annotation, it should not be included.

Comment (me4502) [inline src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java]: I'd probably just say remove them from this PR for now.

File: src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        this.name = name;
+    }
+
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java]: Why do you override `equals` here? In most cases our "fake enum types" should be also checkable with reference equality - that means there is no need to override this method.

File: src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        return null;
+    }
+
+    public void serialize(DataSource source) {
+        // TODO
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/mod/entity/SpongeEntityInteractionType.java]: We don't have `MoreObjects` on 1.8, use `Objects.toStringHelper` instead

**Comment by Aaron1011** [comment]
Closed in https://github.com/SpongePowered/Sponge/commit/a9d4cab22bf394d77b6f9aa175279afa65b52326

**Pull request closed without merging**